app/src/main/java/nerd/tuxmobil/fahrplan/congress/alarms/AlarmReceiver.kt

```diff
@@ -0,0 +1,211 @@
+        private const val INVALID_NOTIFICATION_ID = -1
+        private const val DEFAULT_REQUEST_CODE = 0
+
+        /**
+         * Returns a unique [Intent] to delete the data associated with the
+         * given session alarm `notificationId`.
+         * The `Intent` is composed with a fake data URI to comply with the uniqueness rules
+         * defined by [Intent.filterEquals].
+         */
+        private fun createDeleteNotificationIntent(context: Context, notificationId: Int) =
+            Intent(context, AlarmReceiver::class.java).apply {
+                action = ALARM_DISMISSED
+                putExtra(BUNDLE_KEY_NOTIFICATION_ID, notificationId)
+                data = Uri.parse("fake://$notificationId")
+            }
+
+        private fun getAddAlarmIntent(
+            context: Context,
+            sessionId: String,
+            day: Int,
+            title: String,
+            startTime: Long
+        ) = getAlarmIntent(context, sessionId, day, title, startTime, ALARM_SESSION)
+
+        private fun getDeleteAlarmIntent(
+            context: Context,
+            sessionId: String,
+            day: Int,
+            title: String,
+            startTime: Long
+        ) = getAlarmIntent(context, sessionId, day, title, startTime, ALARM_DELETE)
+
+        private fun getAlarmIntent(
+            context: Context,
+            sessionId: String,
+            day: Int,
+            title: String,
+            startTime: Long,
+            intentAction: String
+        ) = Intent(context, AlarmReceiver::class.java).apply {
+            putExtra(BundleKeys.ALARM_SESSION_ID, sessionId)
+            putExtra(BundleKeys.ALARM_DAY, day)
+            putExtra(BundleKeys.ALARM_TITLE, title)
+            putExtra(BundleKeys.ALARM_START_TIME, startTime)
+            action = intentAction
+            data = Uri.parse("alarm://$sessionId")
+        }
+    }
+
+    private val logging = Logging.get()
+
+    override fun onReceive(context: Context, intent: Intent) {
+        logging.d(LOG_TAG, "Received alarm = " + intent.action + ".")
+
+        when (intent.action){
+            ALARM_DISMISSED -> onSessionAlarmNotificationDismissed(intent)
+            ALARM_UPDATE -> start(context)
+            ALARM_SESSION -> {
+                val sessionId = intent.getStringExtra(BundleKeys.ALARM_SESSION_ID)!!
+                val day = intent.getIntExtra(BundleKeys.ALARM_DAY, 1)
+                val start = intent.getLongExtra(BundleKeys.ALARM_START_TIME, System.currentTimeMillis())
+                val title = intent.getStringExtra(BundleKeys.ALARM_TITLE)!!
+                logging.report(LOG_TAG, "sessionId = $sessionId, intent = $intent")
+                //Toast.makeText(context, "Alarm worked.", Toast.LENGTH_LONG).show();
+
+                val uniqueNotificationId = AppRepository.createSessionAlarmNotificationId(sessionId)
+                val notificationHelper = NotificationHelper(context)
+
+                notificationHelper.notify(
```

Comment:
Anti-pattern: Please don't cascade the calls. It becomes hard to see where an error comes from. Please extract local variables like in the Java version.